cpp/src/arrow/util/utf8_util_test.cc

```diff
@@ -416,6 +416,12 @@ TEST(UTF8StringToUTF16, Basics) {
 
   CheckInvalid("\xff");
   CheckInvalid("h\xc3");
+
+  // lone high-code point
+  CheckInvalid("\xed\xa0\x80");
+
+  // lone low-code point
+  CheckInvalid("\xed\0xb0\x81");
```

Comment:
I think this is incorrect
```suggestion
  CheckInvalid("\xed\xb0\x81");
```